src/Rule/StopOnError.php

```diff
@@ -19,37 +18,54 @@
 use Yiisoft\Validator\WhenInterface;
 
 /**
- * Can be used for validation of nested structures.
+ * Applies to a set of rules, runs validation for each one of them in the order they are defined and stops at the rule
+ * where validation failed. In particular, it can be useful for preventing the heavy operations to increase performance.
+ * It can be set like that for a group of ordered rules:
+ *
+ * ```php
+ * $rule = new StopOnError([
+ *      new HasLength(min: 3),
+ *      // This operation executes DB query and thus heavier. It's preferable not to call it if the previous rule did
+ *      not pass the validation.
+ *      new ExistsInDatabase(),
+ * ]);
+ * ```
+ *
+ * Not to be confused with skipping, there is a separate functionality for that, see {@see SkipOnErrorInterface}.
+ *
```

Comment:
```suggestion
 *
 * ```php
 * $rules = [
 *      new SimpleRule1(), // Let's say there is an error.
 *      // Then this rule is skipped completely with all its related rules.
 *      new StopOnError([
 *          new HeavyRule1(), // Skipped.
 *          new HeavyRule2(), // Skipped.
 *     ]),
 *     new SimpleRule2(), // Skipping of intermediate rules depends on `skipOnError` option.
 * ]);
 * ```
 *
 * Use grouping / ordering / `skipOnError` option to achieve the desired effect,
 *
```